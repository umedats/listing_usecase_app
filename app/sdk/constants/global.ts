const APP_NAME = "SellingPartnerAPIListingSampleApplication";
const APP_VERSION = "1.0";
const APP_LANGUAGE = "TypeScript";
const APP_LANGUAGE_VERSION = "5.0";
/**
 * Header data to be passed to all the SP-API.
 */
export const HEADER_DATA = {
  "User-Agent": `${APP_NAME}/${APP_VERSION} (Language=${APP_LANGUAGE}/${APP_LANGUAGE_VERSION})`,
};
