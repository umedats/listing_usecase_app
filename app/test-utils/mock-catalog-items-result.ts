export const MOCK_CATALOG_ITEMS_API_RESPONSE: object = {
  numberOfResults: 1,
  items: [
    {
      asin: "B00004SPEC",
      productTypes: [
        {
          marketplaceId: "ATVPDKIKX0DER",
          productType: "KITCHEN",
        },
      ],
      summaries: [
        {
          marketplaceId: "ATVPDKIKX0DER",
          adultProduct: false,
          autographed: false,
          brand: "KRUPS",
          browseClassification: {
            displayName: "Countertop Blenders",
            classificationId: "289915",
          },
          itemClassification: "BASE_PRODUCT",
          itemName: "Krups 243-70 Power X Plus Combi Blender, DISCONTINUED",
          manufacturer: "KRUPS",
          memorabilia: false,
          modelNumber: "243-70",
          packageQuantity: 1,
          partNumber: "243-70",
          style: "Modern",
          tradeInEligible: false,
          websiteDisplayGroup: "kitchen_display_on_website",
          websiteDisplayGroupName: "Kitchen",
        },
      ],
    },
  ],
};

export const MOCK_CATALOG_ITEMS_API_EMPTY_RESPONSE: object = {
  numberOfResults: 0,
};
