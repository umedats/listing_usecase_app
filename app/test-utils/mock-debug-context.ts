import { SPAPIRequestResponse } from "@/app/model/types";

export const MOCK_SP_API_RESPONSE: SPAPIRequestResponse = {
  apiName: "GetListingsItem",
  apiDocumentationLink: "https://amazon.com",
  request: {
    marketplaceId: "ATVPDKIKX0DER",
    locale: "en_US",
    asin: "A1F83G8C2ARO7P",
  },
  response: {
    item: "Dummy Item",
  },
};
