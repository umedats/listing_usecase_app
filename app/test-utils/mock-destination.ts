import {
  CreateDestinationResponse,
  Destination,
} from "@/app/sdk/notifications";

export const MOCK_DESTINATION: Destination = {
  name: "TestDestination",
  destinationId: "5fe08e61-c5ec-4427-9f25-24a7b630cce2",
  resource: {
    eventBridge: {
      name: "sellingpartnerapi.amazon.com/amzn1.sellerapps.app.15a75829-cd4a-4efc-b947-0cc39d874577",
      region: "us-east-1",
      accountId: "123456",
    },
  },
};

export const MOCK_CREATE_DESTINATION_RESPONSE: CreateDestinationResponse = {
  payload: MOCK_DESTINATION,
  errors: {},
};
