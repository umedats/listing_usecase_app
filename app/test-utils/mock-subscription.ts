import { Subscription } from "@/app/model/types";
import {
  CREATE_SUBSCRIPTION_API_DOC_LINK,
  CREATE_SUBSCRIPTION_API_NAME,
} from "@/app/constants/global";

export const MOCK_SUBSCRIPTION_1: Subscription = {
  subscriptionId: "f8c3de3d-1fea-4d7c-a8b0-29f63c4c3454",
  notificationType: "BRANDED_ITEM_CONTENT_CHANGE",
  eventBusName: "TestEventBus",
  eventBusFilterRule: "SellingPartnerNotificationsAPIRule",
  sqsQueueName: "BrandedItemContentChange",
};

export const MOCK_SUBSCRIPTION_2: Subscription = {
  subscriptionId: "510f082f-a853-4847-acf2-475641cb7b06",
  notificationType: "PRODUCT_TYPE_DEFINITIONS_CHANGE",
  eventBusName: "TestEventBus1",
  eventBusFilterRule: "SellingPartnerNotificationsAPIRule",
  sqsQueueName: "ProductTypeDefinitionsChange",
};

export const MOCK_SUBSCRIPTION_3: Subscription = {
  subscriptionId: "510f082f-a853-4847-acf2-475641cb7b06",
  notificationType: "PRODUCT_TYPE_DEFINITIONS_CHANGE",
  eventBusName: "TestEventBus1",
  eventBusFilterRule: "SellingPartnerNotificationsAPIRule",
  sqsTargetId: "ProductTypeDefinitionsTarget",
  sqsQueueName: "ProductTypeDefinitionsChange",
};

export const MOCK_SUBSCRIPTION_4: Subscription = {
  subscriptionId: "510f082f-a853-4847-acf2-475641cb7b06",
  notificationType: "PRODUCT_TYPE_DEFINITIONS_CHANGE",
  eventBusName: "TestEventBus1",
  eventBusFilterRule: "SellingPartnerNotificationsAPIRule",
};

export const MOCK_CREATE_SUBSCRIPTION_RESULT_1 = {
  subscriptionId: "510f082f-a853-4847-acf2-475641cb7b06",
  payloadVersion: "1.0",
  destinationId: "testDestinationId",
};

export const MOCK_CREATE_SUBSCRIPTION_SP_API_RESPONSE = {
  apiName: CREATE_SUBSCRIPTION_API_NAME,
  apiDocumentationLink: CREATE_SUBSCRIPTION_API_DOC_LINK,
  request: {
    body: {
      payloadVersion: "1",
      destinationId: "testDestinationId",
    },
    notificationType: "LISTINGS_ITEM_STATUS_CHANGE",
  },
  response: MOCK_CREATE_SUBSCRIPTION_RESULT_1,
};
