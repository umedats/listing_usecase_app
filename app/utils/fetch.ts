import { SPAPIRequestResponse } from "@/app/model/types";
import { SettableDebugState } from "@/app/context/debug-context-provider";
import { NO_CACHING_DIRECTIVE } from "@/app/constants/global";

/**
 * Helper function to fetch the data from the backend API. This function
 * extracts the debug context, and invokes the updateDebugContextCallback
 * method.
 * @param url the url of the backend API.
 * @param updateDebugContextCallback a callback to update the debug context.
 * @param responseDataCallback an optional call back which is invoked with the
 * data extracted from the backend API.
 * @param requestInit params to initialize the fetch request.
 */
export async function fetcherForBackendAPI(
  url: string,
  updateDebugContextCallback: (debugData: any) => void,
  responseDataCallback?: (data: any) => any | void,
  requestInit?: RequestInit,
) {
  let newRequestInit = requestInit;
  if (requestInit) {
    newRequestInit = { ...requestInit, cache: NO_CACHING_DIRECTIVE };
  }

  const response = await fetch(url, newRequestInit);
  const responseBody = await response.json();
  if (responseBody.debugContext) {
    updateDebugContextCallback(responseBody.debugContext);
  }

  if (!response.ok) {
    throw [response.status, response.statusText];
  }

  const data = responseBody.data;
  if (responseDataCallback) {
    return responseDataCallback(data);
  }
  return data;
}

/**
 * Saves the given debugData for the given route into debugContext.
 * @param pathname
 * @param debugData
 * @param debugContext
 */
export function updateDebugContext(
  pathname: string,
  debugData: any,
  debugContext: SettableDebugState,
) {
  (debugData as SPAPIRequestResponse[]).forEach((value) =>
    debugContext.addRequestResponse(pathname, value),
  );
}
