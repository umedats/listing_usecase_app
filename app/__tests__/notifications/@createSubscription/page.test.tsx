import CreateSubscription from "@/app/[locale]/notifications/@createSubscription/page";
import { cleanup, render, screen, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import "@testing-library/react";
import { NextIntlProvider } from "next-intl";
import translations from "@/app/internationalization/translations/en-US.json";
import {
  mockRejectFetchResponse,
  mockResolveFetchResponse,
} from "@/app/test-utils/mock-fetch";
import userEvent from "@testing-library/user-event";
import fetch from "jest-fetch-mock";
import { MOCK_SP_API_RESPONSE } from "@/app/test-utils/mock-debug-context";
import { MOCK_CREATE_SUBSCRIPTION_RESULT_1 } from "@/app/test-utils/mock-subscription";
import { US_LOCALE } from "@/app/constants/global";

function renderPage() {
  const { asFragment, queryByText, getByText } = render(
    <NextIntlProvider locale={US_LOCALE} messages={translations}>
      <CreateSubscription />
    </NextIntlProvider>,
  );

  return { asFragment, queryByText, getByText };
}

describe("Tests for the CreateSubscription", () => {
  beforeEach(() => {
    fetch.resetMocks();
  });

  afterEach(() => {
    // Cleanup any resources or state after each test
    cleanup();
  });

  test("Snapshot Test", async () => {
    const { asFragment } = renderPage();
    expect(asFragment()).toMatchSnapshot();
  });

  test("CreateSubscription Error Test", async () => {
    mockRejectFetchResponse(500);
    renderPage();

    await userEvent.click(screen.getByTestId("submitCreateSubscription"));
    await waitFor(() => {
      const dialog = screen.queryByText("Please try again", { exact: false });
      expect(dialog).not.toBeNull();
      expect(dialog).toBeVisible();
    });
    expect(screen.queryByTestId("alertDialog")).toMatchSnapshot();
    await userEvent.click(screen.getByTestId("cancelAlertDialog"));
    await waitFor(() => expect(screen.queryByTestId("alertDialog")).toBeNull());
  });

  test("CreateSubscription Success Test", async () => {
    mockResolveFetchResponse(200, {
      data: MOCK_CREATE_SUBSCRIPTION_RESULT_1,
      debugContext: [MOCK_SP_API_RESPONSE],
    });
    renderPage();

    await userEvent.click(screen.getByTestId("submitCreateSubscription"));
    await waitFor(() => {
      const dialog = screen.queryByText("Created subscription successfully", {
        exact: false,
      });
      expect(dialog).not.toBeNull();
      expect(dialog).toBeVisible();
    });

    expect(screen.queryByTestId("alertDialog")).toMatchSnapshot();

    await userEvent.click(screen.getByTestId("cancelAlertDialog"));
    await waitFor(() => expect(screen.queryByTestId("alertDialog")).toBeNull());
  });

  test("CreateSubscription Success Test For Conflicting resource", async () => {
    mockResolveFetchResponse(200, {
      data: {},
      debugContext: [MOCK_SP_API_RESPONSE],
    });
    renderPage();

    await userEvent.click(screen.getByTestId("submitCreateSubscription"));
    await waitFor(() => {
      const dialog = screen.queryByText("Subscription already exists", {
        exact: false,
      });
      expect(dialog).not.toBeNull();
      expect(dialog).toBeVisible();
    });

    expect(screen.queryByTestId("alertDialog")).toMatchSnapshot();

    await userEvent.click(screen.getByTestId("cancelAlertDialog"));
    await waitFor(() => expect(screen.queryByTestId("alertDialog")).toBeNull());
  });
});
