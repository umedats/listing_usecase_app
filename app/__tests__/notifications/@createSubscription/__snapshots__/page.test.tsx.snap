// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the CreateSubscription CreateSubscription Error Test 1`] = `
<div
  class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
  data-testid="alertDialog"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-testid="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    role="presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-labelledby=":r6:"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
        id=":r6:"
      >
        Error
      </h2>
      <div
        class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
      >
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiDialogContentText-root css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          There was unexpected error happened while creation of the subscription. Please try again
        </p>
      </div>
      <div
        class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-testid="cancelAlertDialog"
          tabindex="0"
          type="button"
        >
          Close
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    data-testid="sentinelEnd"
    tabindex="0"
  />
</div>
`;

exports[`Tests for the CreateSubscription CreateSubscription Success Test 1`] = `
<div
  class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
  data-testid="alertDialog"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-testid="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    role="presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-labelledby=":ra:"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
        id=":ra:"
      >
        Success
      </h2>
      <div
        class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
      >
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiDialogContentText-root css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          Created subscription successfully! The subscription ID is 510f082f-a853-4847-acf2-475641cb7b06, payload version is 1.0, and destination ID is testDestinationId.
        </p>
      </div>
      <div
        class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-testid="cancelAlertDialog"
          tabindex="0"
          type="button"
        >
          Close
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    data-testid="sentinelEnd"
    tabindex="0"
  />
</div>
`;

exports[`Tests for the CreateSubscription CreateSubscription Success Test For Conflicting resource 1`] = `
<div
  class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
  data-testid="alertDialog"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-testid="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    role="presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-labelledby=":re:"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
        id=":re:"
      >
        Error
      </h2>
      <div
        class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
      >
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiDialogContentText-root css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          Subscription already exists for the notification type. Please delete the associated subscription for this notification type before creating another one.
        </p>
      </div>
      <div
        class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-testid="cancelAlertDialog"
          tabindex="0"
          type="button"
        >
          Close
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    data-testid="sentinelEnd"
    tabindex="0"
  />
</div>
`;

exports[`Tests for the CreateSubscription Snapshot Test 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-13mqtjd-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          Create Listing Subscription
        </h6>
      </div>
    </div>
    <form
      data-testid="createSubscriptionRequestSubmit"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item css-1f064cs-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-18ksmt-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
            >
              <label
                aria-label="The drop down lists all the supported notification types which belong to Listings Domain."
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
                data-mui-internal-clone-element="true"
                for="notificationTypeDropDown"
              >
                Subscriptions
              </label>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9 css-1qpzhgb-MuiGrid-root"
            >
              <div
                aria-label="The drop down lists all the supported notification types which belong to Listings Domain."
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-sizeSmall  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                data-mui-internal-clone-element="true"
                data-testid="notificationTypeDropDown"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="notificationTypeDropDown"
                  class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-jedpe8-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                  id="notificationTypeDropDown"
                  role="button"
                  tabindex="0"
                >
                  LISTINGS_ITEM_ISSUES_CHANGE
                </div>
                <input
                  aria-hidden="true"
                  aria-invalid="false"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  tabindex="-1"
                  value="LISTINGS_ITEM_ISSUES_CHANGE"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item css-1f064cs-MuiGrid-root"
        >
          <span
            aria-label="Click this button to create a subscription"
            class=""
            data-mui-internal-clone-element="true"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
              data-testid="submitCreateSubscription"
              id="submitCreateSubscription"
              tabindex="0"
              type="button"
            >
              Create Subscription
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </span>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
